File: NumberGuessingGame.clj
(ns number-guessing-game)


(defn guessed-wrong [hint]
    (println (str "Sorry, guess again but " hint))
    false)


(defn guessed-right [guessCount]
    (println (str "You win! It took you " guessCount " attempts."))
    true)


(defn evaluate [num guess guessCount]
    (cond
        (> num guess) (guessed-wrong "higher.")
        (< num guess) (guessed-wrong "lower.")
    :else (guessed-right guessCount)))


(defn game
    []
    (println (str "Guess the random number between 1 - 100!")))
    (let [num (inc (rand-int 100))]
        (loop [guessCount 1]
            (if (= true (evaluate num(let [guess (read)]) guessCount))
                -1
                (recur (inc guessCount))))))
(game)

File: NumberGuessingGame.java
import java.util.Random;
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Random randNum = new Random();
        int num = randNum.nextInt(100);
        int guessCount = 0;
        Scanner scan = new Scanner(System.in);
        int guess;
        boolean win = false;

        while(win == false) {
            System.out.println("Guess a number between 1 to 100: ");
            guess = scan.nextInt();
            guessCount++;

            if(guess == num) {
                win = true;
            }
            else if(guess < num){
                System.out.println("Your guess is lower than the correct number!");
            }
            else if(guess > num) {
                System.out.println("Your guess is higher than the correct number!");
            }
        }
        System.out.println("You won!");
        System.out.println("You needed " + guessCount + " attempts to guess the right number");
    }
}